virtual_rainforest/main.py

```diff
@@ -171,6 +173,85 @@ def extract_timing_details(
     return start_time, update_interval, raw_interval, end_time
 
 
+def output_current_state(
+    data: Data,
+    models_cfd: dict[str, BaseModel],
+    data_options: dict[str, Any],
+    time_index: int,
+) -> None:
+    """Function to output the current state of the data object.
+
+    This function outputs all variables stored in the data object, except for any data
+    with a "time_index" dimension defined (at present only climate input data has this).
+    This data can either be saved as a new file or appended to an existing file.
+
+    Args:
+        data: Data object to output current state of
+        models_cfd: Set of models configured for use in the simulation
+        data_options: Set of options concerning what to output and where
+        time_index: The index representing the current time step in the data object.
+
+    Raises:
+        ConfigurationError: If the final output directory doesn't exist, isn't a
+           directory, or the final output file already exists (when in new file mode).
+           If the file to append to is missing (when not in new file mode).
+    """
+
+    # Only variables in the data object that are updated by a model should be outputted
+    all_variables = [
+        models_cfd[model_nm].vars_updated for model_nm in models_cfd.keys()
+    ]
+    # Then flatten the list
+    variables_to_save = [item for sublist in all_variables for item in sublist]
+    # Create output file path for specific time index
+    out_path_name = (
+        f"{data_options['out_folder_continuous']}/continuous_state{time_index}.nc"
+    )
+
+    # Save the required variables by appending to existing file
+    data.save_timeslice_to_netcdf(Path(out_path_name), variables_to_save, time_index)
+
+
+def merge_continuous_data_files(data_options: dict[str, Any]) -> None:
+    """Merge all continuous data files in a folder into a single file.
+
+    It is important to note that this function deletes all of the original continuos
+    output files once the combined output is saved. It also combines (and deletes) all
+    files matching the pattern 'continuous_state*.nc' that it finds in the specified
+    output folder.
+
+    Args:
+        data_options: Set of options concerning what to output and where
+
+    Raises:
+        ConfigurationError: If output folder doesn't exist or if it output file already
+            exists
+    """
+    # Find Path of folder containing the continuous output
+    out_folder = Path(data_options["out_folder_continuous"])
+    # Find all relevant files in the given folder and convert to list
+    continuous_files = out_folder.rglob("continuous_state*.nc")
+
+    # Open all files as a single dataset
+    all_data = open_mfdataset(list(continuous_files))
+
+    # Check that output file doesn't already exist
+    out_path = Path(f"{out_folder}/all_continuous_data.nc")
+    try:
+        check_outfile(out_path)
```

Comment:
Maybe this check should be done before collecting all files, otherwise among the collected files you will have one containing all collected data. That will save you from having to close `all_data` as it won't be open, to start with.